### PR TITLE
Replace npm in dev bundle with version from our own fork

### PR DESCRIPTION
**Pull request by martijnwalraven**

### CODE REVIEW

File: scripts/generate-dev-bundle.sh [hunk 1]
```diff
@@ -167,6 +173,22 @@ mv BrowserStackLocal "$DIR/bin/"
...
 rm -rf "$DIR/bin/npm3"
 
+# Sanity check to see if we're not breaking anything by replacing npm
+INSTALLED_NPM_VERSION=$(cat "$DIR/lib/node_modules/npm/package.json" |
+xargs -0 node -e "console.log(JSON.parse(process.argv[1]).version)")
+if [ "$INSTALLED_NPM_VERSION" != "1.3.4" ]; then
+  echo "Unexpected NPM version in lib/node_modules: $INSTALLED_NPM_VERSION"
+  echo "We will be replacing it with our own version because the bundled node"
+  echo "is built using PORTABLE=1, which makes npm look for node relative to"
+  echo "it's own directory."
```
Comment (glasser) [inline scripts/generate-dev-bundle.sh]: "its"

File: scripts/generate-dev-bundle.sh [hunk 2]
```diff
@@ -67,6 +67,12 @@ else
     curl "${MONGO_URL}" | tar zx
 fi
 
+# Copy bundled npm to temporary directory so we can restore it later
+# We do this because the bundled node is built using PORTABLE=1,
+# which makes npm look for node relative to it's own directory
+# See build-node-for-dev-bundle.sh
+cp -r "$DIR/lib/node_modules/npm" "$DIR/bundled-npm"
```
Comment (glasser) [inline scripts/generate-dev-bundle.sh]: maybe mv?  since we're explicitly not using this one anyway?

Comment (martijnwalraven) [inline scripts/generate-dev-bundle.sh]: Well, we still need it to be there to run `npm install npm@3.1.2` I think?

Comment (glasser) [inline scripts/generate-dev-bundle.sh]: Good point!

**Pull request merged**